ge/org.osate.ge/src/org/osate/ge/internal/ui/properties/SetDimensionsPropertySection.java

```diff
@@ -214,11 +211,16 @@ private ArrayDimension getSelectedDimension() {
 				@Override
 				public void widgetSelected(final SelectionEvent e) {
 					selectedIndex = tableViewer.getTable().getSelectionIndex();
-					final ArrayDimension dim = getSelectedDimension();
-					if (dim != null && modifiedArrayDimension(dim, SelectionUtil.getProjectOrThrow(dim.eResource()))) {
+					final ArrayDimension readonlyDim = getSelectedDimension();
+					if (readonlyDim != null) {
 						final Aadl2Package pkg = Aadl2Factory.eINSTANCE.getAadl2Package();
-						selectedBos.modify(ArrayableElement.class, ae -> {
-							ae.getArrayDimensions().set(selectedIndex, createArrayDimensionDuplicate(pkg, dim));
+						final ArrayDimension dim = createArrayDimensionDuplicate(pkg, readonlyDim);
+						selectedBos.boStream(ArrayableElement.class).findAny().ifPresent(ele -> {
+							if (modifiedArrayDimension(dim, SelectionUtil.getProjectOrThrow(ele.eResource()))) {
+								selectedBos.modify(ArrayableElement.class, ae -> {
+									ae.getArrayDimensions().set(selectedIndex, createArrayDimensionDuplicate(pkg, dim));
```

Comment:
I think this should change to "ae.getArrayDimensions().set(selectedIndex, dim);"
instead of creating array dimension duplicate again.